### ISSUE TITLE
NPE in response.body when content type lacks charset

### ISSUE THREAD

**Issue by maxcom** [playframework/play-ws#182]
`response.body` fails with NPE when response type is `application/json`:

```
NullPointerException: charset java.lang.NullPointerException: charset java.lang.NullPointerException: charset
at java.lang.String.<init>(String.java:461)
at java.lang.String.<init>(String.java:515)
at play.api.libs.ws.ahc.cache.CacheableResponse.getResponseBody(CacheableResponse.scala:132)
at play.api.libs.ws.ahc.cache.CacheableResponse.getResponseBody(CacheableResponse.scala:124)
at play.api.libs.ws.ahc.StandaloneAhcWSResponse.body$lzycompute(StandaloneAhcWSResponse.scala:43)
at play.api.libs.ws.ahc.StandaloneAhcWSResponse.body(StandaloneAhcWSResponse.scala:38)
at play.api.libs.ws.ahc.AhcWSResponse.body(AhcWSResponse.scala:64)
```

play-ws version 1.0.1

**Comment by marcospereira** [comment]
Thank you, @maxcom.

See #183 for a fix.

**Comment by marcospereira** [comment]
@maxcom could you please verify that version 1.0.4 solves this problem? You can add the following dependencies (temporarily) to test the new version:

```scala
libraryDependencies ++= Seq(
  "com.typesafe.play" %% "play-ws-standalone" % "1.0.4",
  "com.typesafe.play" %% "play-ws-standalone-xml" % "1.0.4",
  "com.typesafe.play" %% "play-ws-standalone-json" % "1.0.4",
  "com.typesafe.play" %% "play-ahc-ws-standalone" % "1.0.4",
  "com.typesafe.play" % "shaded-asynchttpclient" % "1.0.4",
  "com.typesafe.play" % "shaded-oauth" % "1.0.4"
)
```

**Comment by maxcom** [comment]
Thank you. Code looks good. I can't check the fix, compilation fails when I add that dependencies:

```
[error] /home/maxcom/git-checkout/dozor-httpd/app/service/ws/ProxyService.scala:55: Cannot find an instance of akka.stream.scaladsl.Source[Product with Serializable with play.api.mvc.MultipartFormData.Part[akka.stream.scaladsl.Source[akka.util.ByteString,scala.concurrent.Future[akka.stream.IOResult]]],akka.NotUsed] to WSBody. Define a BodyWritable[akka.stream.scaladsl.Source[Product with Serializable with play.api.mvc.MultipartFormData.Part[akka.stream.scaladsl.Source[akka.util.ByteString,scala.concurrent.Future[akka.stream.IOResult]]],akka.NotUsed]] or extend play.api.libs.ws.ahc.DefaultBodyWritables
[error]       baseRequest.withBody(Source(parts))
[error]                           ^
```

**Comment by marcospereira** [comment]
Thanks for the feedback, @maxcom. Could you please share the code where compilation is failing and also your build.sbt file?

**Comment by maxcom** [comment]
Look here: https://github.com/maxcom/play-compile-problem (this project is made only to show compile problem, it is not made to run)

**Comment by marcospereira** [comment]
Hey @maxcom,

this is not related to the original issue reported here. To solve the compilation problem, you need an implicit `BodyWritable[JsValue]` available, so add the following import to your class:

```scala
import play.api.libs.ws.JsonBodyWritables._
```

This should solve the compilation problem. There is a number of `BodyWriteables` available by default. See the following scaladocs:

https://www.playframework.com/documentation/2.6.x/api/scala/play/api/libs/ws/WSBodyWritables.html

You may also be interested in `BodyReadables`:

https://www.playframework.com/documentation/2.6.x/api/scala/play/api/libs/ws/WSBodyReadables.html

**Comment by maxcom** [comment]
Yes, I'm sure that it is not related to this bug.

It is strange that I need to add new import line when I add 1.0.4 to dependency list. It is simplified code. In actual project I had to add `import play.api.libs.ws.WSBodyWritables._` to large number of source files.

**Comment by maxcom** [comment]
Original problem is fixed in 1.0.4.
